transfermethodrepository/src/main/java/com/hyperwallet/android/ui/transfermethod/repository/TransferMethodUpdateConfigurationRepositoryImpl.java

```diff
@@ -0,0 +1,118 @@
+import com.hyperwallet.android.ui.common.repository.EspressoIdlingResource;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TransferMethodUpdateConfigurationRepositoryImpl implements TransferMethodUpdateConfigurationRepository {
+    private final Handler mHandler;
+    private final Map<FieldMapKey, HyperwalletTransferMethodConfigurationField> mFieldMap;
+    private HyperwalletTransferMethodConfigurationKey mTransferMethodConfigurationKey;
+
+    //todo use default modifier after RepositoryFactory is removed
+    public TransferMethodUpdateConfigurationRepositoryImpl() {
+        mHandler = new Handler();
+        mFieldMap = new HashMap<>();
+    }
+
+    @VisibleForTesting()
+    protected TransferMethodUpdateConfigurationRepositoryImpl(@Nullable Handler handler,
+            HyperwalletTransferMethodConfigurationKey transferMethodConfigurationKey,
+            Map<FieldMapKey, HyperwalletTransferMethodConfigurationField> fieldMap) {
+        mHandler = handler;
+        mTransferMethodConfigurationKey = transferMethodConfigurationKey;
+        mFieldMap = fieldMap;
+    }
+
+    @VisibleForTesting
+    Hyperwallet getHyperwallet() {
+        return Hyperwallet.getDefault();
+    }
+
+    @VisibleForTesting
+    void getTransferMethodConfigurationFieldResult(
+            @NonNull final String transferMethodType,
+            @NonNull final String transferMethodToken,
+            @NonNull final LoadFieldsCallback loadFieldsCallback) {
+        TransferMethodUpdateConfigurationFieldQuery query = new TransferMethodUpdateConfigurationFieldQuery(
+                transferMethodToken);
+        EspressoIdlingResource.increment();
+
+        getHyperwallet().retrieveUpdateTransferMethodConfigurationFields(
+                query,
+                new HyperwalletListener<HyperwalletTransferMethodConfigurationField>() {
+                    @Override
+                    public void onSuccess(HyperwalletTransferMethodConfigurationField result) {
+                        FieldMapKey fieldMapKey = new FieldMapKey(transferMethodType);
+                        mFieldMap.put(fieldMapKey, result);
+                        loadFieldsCallback.onFieldsLoaded(result);
+                        EspressoIdlingResource.decrement();
+                    }
+
+                    @Override
+                    public void onFailure(HyperwalletException exception) {
+                        loadFieldsCallback.onError(exception.getErrors());
+                        EspressoIdlingResource.decrement();
+                    }
+
+                    @Override
+                    public Handler getHandler() {
+                        return mHandler;
+                    }
+                });
+
+    }
+
+    @Override
+    public synchronized void getFields(@NonNull final String transferMethodType,
+            @NonNull final String transferMethodToken,
+            @NonNull final LoadFieldsCallback loadFieldsCallback) {
```

Comment:
transferMethodType not needed for graphql call

Reply:
removed